### PR TITLE
added livestreams to getChannelHome

### PR DESCRIPTION
**Pull request by free-bots**
Channels with livestreams on the main page caused crashes with getChannelHome. I added some checks for the layout and the new livestream elements are now part of items.

### CODE REVIEW

File: app/helper.js [hunk 1]
```diff
@@ -192,7 +192,10 @@ class YoutubeGrabberHelper {
     if (typeof (obj.richItemRenderer) !== 'undefined') {
       video = obj.richItemRenderer.content.videoRenderer
       video.lengthSeconds = video.lengthText.simpleText.split(':').reduce((acc, time) => (60 * acc) + +time)
-      video.title.simpleText = video.title.runs[0].text
+      video.title.simpleText = video.title.runs.at(0).text
```
Comment (ChunkyProgrammer) [inline app/helper.js]:
```suggestion
      video.title.simpleText = video.title.runs[0].text
```

File: app/helper.js [hunk 2]
```diff
@@ -192,7 +192,10 @@ class YoutubeGrabberHelper {
...
+    } else if ((typeof obj.videoRenderer) !== 'undefined') {
+      video = obj.videoRenderer
+      video.title.simpleText = video.title.runs.at(0).text
```
Comment (ChunkyProgrammer) [inline app/helper.js]:
```suggestion
      video.title.simpleText = video.title.runs[0].text
```

File: app/helper.js [hunk 3]
```diff
@@ -567,65 +577,82 @@ class YoutubeGrabberHelper {
...
-
-      if ('runs' in shelf.title) {
-        const title = shelf.title.runs[0]
-        if ('navigationEndpoint' in title) {
-          shelfUrl = title.navigationEndpoint.commandMetadata.webCommandMetadata.url
-        }
-        shelfName = title.text
+      if ('channelFeaturedContentRenderer' in item.itemSectionRenderer.contents[0]) {
+        type = 'livestreams'
+        shelfName = item.itemSectionRenderer.contents[0].channelFeaturedContentRenderer.title.runs.map(run => run.text).join(" ");
```
Comment (ChunkyProgrammer) [inline app/helper.js]:
```suggestion
        shelfName = item.itemSectionRenderer.contents[0].channelFeaturedContentRenderer.title.runs.map(run => run.text).join(' ')
```

**Review by ChunkyProgrammer — COMMENTED** [review]
Hi, I enabled workflows on your PRs 😄 . Would you be able to fix the small lint issue (use single quotes instead of double quotes and remove the semicolon). Other than that, this pr looks good to me.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Review by ChunkyProgrammer — APPROVED** [review]

**Review by PikachuEXE — APPROVED** [review]

**Pull request merged**